### PR TITLE
remove t0 parameter from matrixfit and extract it from the correlator…

### PR DESCRIPTION
**Pull request by kostrzewa**
… metadata

### CODE REVIEW

**Comment by kostrzewa** [comment]
@martin-ueding can you test this in your analysis ?

File: R/cf.R [hunk 1]
```diff
@@ -166,7 +166,7 @@ cf_principal_correlator <- function (.cf = cf(), id, gevp_reference_time) {
   .cf$id <- id
   .cf$gevp_reference_time <- gevp_reference_time
 
-  class(.cf) <- append(class(.cf), 'cf_principal_correlators')
```
Comment (kostrzewa) [inline R/cf.R]: good catch @urbach

Comment (kostrzewa) [reply R/cf.R]: @martin-ueding do you agree with this change? since the mixin-constructor is singular, the name of the mixin should be too, right?

**Comment by urbach** [comment]
> kostrzewa commented on this pull request.
> 
> 
> 
> > @@ -166,7 +166,7 @@ cf_principal_correlator <- function (.cf = cf(), id, gevp_reference_time) {
>    .cf$id <- id
>    .cf$gevp_reference_time <- gevp_reference_time
>  
> -  class(.cf) <- append(class(.cf), 'cf_principal_correlators')
> 
> good catch @urbach 

well, I catched it because the `bootstrap.gevp` example failed due
to this. Writing examples is really helpful! ;) There is a function
"run_examples" in the devtools package...

**Pull request merged**